src/drivers/snapdragon_pwm_out/snapdragon_pwm_out.cpp

```diff
@@ -0,0 +1,608 @@
+
+void pwm_deinitialize();
+
+void send_outputs_pwm(const uint16_t *pwm);
+
+void task_main_trampoline(int argc, char *argv[]);
+
+void subscribe();
+
+void task_main(int argc, char *argv[]);
+
+int initialize_mixer(const char *mixer_filename);
+
+int mixer_control_callback(uintptr_t handle, uint8_t control_group, uint8_t control_index, float &input);
+
+
+
+/*
+ * functions
+ */
+
+int mixer_control_callback(uintptr_t handle,
+			   uint8_t control_group,
+			   uint8_t control_index,
+			   float &input)
+{
+	const actuator_controls_s *controls = (actuator_controls_s *)handle;
+
+	input = controls[control_group].control[control_index];
+
+	return 0;
+}
+
+
+int initialize_mixer(const char *mixer_filename)
+{
+
+	char buf[2048];
+	size_t buflen = sizeof(buf);
+	PX4_INFO("Trying to initialize mixer from config file %s", mixer_filename);
+	int fd_load = ::open(mixer_filename, O_RDONLY);
+	if (fd_load != -1) {
+		int nRead = ::read(fd_load, buf, buflen);
+		close(fd_load);
+
+		if (nRead > 0) {
+			_mixer = MultirotorMixer::from_text(mixer_control_callback, (uintptr_t)&_controls, buf, buflen);
+
+			if (_mixer != nullptr) {
+				PX4_INFO("Successfully initialized mixer from config file");
+				return 0;
+
+			} else {
+				PX4_ERR("Unable to parse from mixer config file");
+				return -1;
+			}
+
+		} else {
+			PX4_WARN("Unable to read from mixer config file");
+			return -2;
+		}
+
+	} else {
+		PX4_WARN("No mixer config file found, using default mixer.");
+
+		/* Mixer file loading failed, fall back to default mixer configuration for
+		* QUAD_X airframe. */
+		float roll_scale = 1;
+		float pitch_scale = 1;
+		float yaw_scale = 1;
+		float deadband = 0;
+
+		_mixer = new MultirotorMixer(mixer_control_callback, (uintptr_t)&_controls,
+					     MultirotorGeometry::QUAD_X,
+					     roll_scale, pitch_scale, yaw_scale, deadband);
+
+		if (_mixer == nullptr) {
+			return -1;
+		}
+
+		return 0;
+
+	}
+
+}
+
+void subscribe()
+{
+	memset(_controls, 0, sizeof(_controls));
+	memset(_poll_fds, 0, sizeof(_poll_fds));
+
+	/*
+	 * ORB topic names to msg actuator_controls
+	 */
+	_controls_topics[0] = ORB_ID(actuator_controls_0);
+	_controls_topics[1] = ORB_ID(actuator_controls_1);
+	_controls_topics[2] = ORB_ID(actuator_controls_2);
+	_controls_topics[3] = ORB_ID(actuator_controls_3);
+
+
+	/*
+	 * subscibe to orb topic
+	 */
+	for (uint8_t i = 0; i < actuator_controls_s::NUM_ACTUATOR_CONTROL_GROUPS; i++) {
+
+		if (_groups_required & (1 << i)) {
+
+			PX4_DEBUG("subscribe to actuator_controls_%d", i);
+
+			_controls_subs[i] = orb_subscribe(_controls_topics[i]);
+
+		} else {
+
+			_controls_subs[i] = -1;
+
+		}
+
+		if (_controls_subs[i] >= 0) {
+
+			_poll_fds[_poll_fds_num].fd = _controls_subs[i];
+			_poll_fds[_poll_fds_num].events = POLLIN;
+			_poll_fds_num++;
+
+		}
+
+		_armed_sub = orb_subscribe(ORB_ID(actuator_armed));
+
+	}
+
+}
+
+
+
+int pwm_initialize(const char *device)
+{
+	/*
+	 * open PWM device
+	 */
+	int i = 0;
```

Comment:
Please move the `int` declaration into the for scope. Otherwise, you can shoot yourself in the foot with shadowing variables.